core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -107,6 +108,35 @@
   private SqlFunctions() {
   }
 
+  /** SQL TO_BASE64(string) function*/
+  public static String toBase64(String s) {
+    if (s == null) {
+      return null;
+    }
+    try {
+      return Base64.getEncoder().encodeToString(s.getBytes("UTF-8"));
+    } catch (Exception e) {
+      return null;
+    }
+  }
+
+  /** SQL FROM_BASE64(string) function*/
+  public static  String fromBase64(String s) {
+    if (s == null) {
+      return null;
+    }
+    try {
+      StringBuilder builder = new StringBuilder();
+      byte[] afterDecode = Base64.getDecoder().decode(s);
+      for (byte b : afterDecode) {
+        builder.append((char) b);
+      }
+      return builder.toString();
+    } catch (Exception e) {
```

Comment:
[decode](https://docs.oracle.com/javase/8/docs/api/java/util/Base64.Decoder.html#decode-java.lang.String-) throws `IllegalArgumentException` maybe catch it instead of generic exception ?